scripts/accounts-invariants/pallet-balances-maintenance.py

```diff
@@ -599,6 +617,127 @@ def set_providers_counter_to_one(chain_connection, account_info, internal_system
+        reserved = account_info['data']['reserved']
+        if consumers == 0 and reserved > 0:
+            zero_consumers_underflow_accounts.append(account_id_and_info)
+            log.debug(f"Found an account that has zero consumers and non-zero reserved: {account_id_and_info}")
+            continue
+        if account_id in locks and len(locks[account_id]) > 0:
+            log.debug(f"Account {account_id} has following locks: {locks[account_id]}")
+            vesting_lock = next(filter(lambda lock: lock['id'] == "0x76657374696e6720", locks[account_id]), None)
+            if vesting_lock is not None and consumers == 1:
+                log.debug(f"Found vesting lock: {vesting_lock}")
+                zero_consumers_underflow_accounts.append(account_id_and_info)
+                log.debug(f"Found an account that has one consumer and vesting lock: {account_id_and_info}")
+                continue
+            staking_lock = next(filter(lambda lock: lock['id'] == "0x7374616b696e6720", locks[account_id]), None)
+            if staking_lock is not None:
+                log.debug(f"Found staking lock: {staking_lock}")
+                if consumers == 2:
+                    zero_consumers_underflow_accounts.append(account_id_and_info)
+                    log.debug(f"Found an account that has two consumers and staking lock: {account_id_and_info}")
+                    continue
+                if consumers == 3 and account_id in next_keys:
+                    zero_consumers_underflow_accounts.append(account_id_and_info)
+                    log.debug(f"Found an account that has three consumers, staking lock, and next session key: "
+                              f"{account_id_and_info}")
+                    continue
+    return zero_consumers_underflow_accounts
+
+
+def batch_fix_accounts_consumers_underflow(chain_connection,
+                                           input_args,
+                                           accounts,
+                                           sender_keypair):
+    """
+    Send Operations.fix_accounts_consumers_underflow call in a batch
+    :param chain_connection: WS connection handler
+    :param input_args: script input arguments returned from argparse
+    :param accounts: list of accounts to fix their consumers counter
+    :param sender_keypair: keypair of sender account
+    :return: None. Can raise exception in case of SubstrateRequestException thrown
+    """
+    for (i, account_ids_chunk) in enumerate(chunks(accounts, input_args.fix_consumers_calls_in_batch)):
+        operations_calls = list(map(lambda account: chain_connection.compose_call(
+            call_module='Operations',
+            call_function='fix_accounts_consumers_underflow',
+            call_params={
+                'who': account
+            }), account_ids_chunk))
+        batch_call = chain_connection.compose_call(
+            call_module='Utility',
+            call_function='batch',
```

Comment:
`batch` is generally a suboptimal choice here (and in most cases), either use:
- `batch_all` if you want to make sure that either all succeed or none
- or use `force_batch` if you want as many succeeded as possible

`batch` is somewhere in the middle, it stops on first error, but does not revert

Reply:
Good spot, I'll use `batch_all`